frontend/cypress/integration/common/sidecar_injection.ts

```diff
@@ -0,0 +1,274 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+
+Given('a namespace without override configuration for automatic sidecar injection', function () {
+    this.targetNamespace = "default";
+
+    // Make sure that the target namespace does not have override configuration
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": null,
+                "istio.io/rev": null
+            }
+        }
+    });
+});
+
+Given('a namespace which has override configuration for automatic sidecar injection', function () {
+    this.targetNamespace = "default";
+    this.istioInjection = "enabled";
+
+    // Make sure that the target namespace has some override configuration
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": this.istioInjection,
+                "istio.io/rev": null
+            }
+        }
+    });
+});
+
+Given('the override configuration for sidecar injection is {string}', function (enabledOrDisabled) {
+    if (this.istioInjection !== enabledOrDisabled) {
+        cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+            metadata: {
+                labels: {
+                    "istio-injection": enabledOrDisabled,
+                    "istio.io/rev": null
+                }
+            }
+        });
+        this.istioInjection = enabledOrDisabled;
+    }
+});
+
+Given('a workload without a sidecar', function () {
+    this.targetNamespace = 'default';
+    this.targetWorkload = 'sleep';
+
+    // To achieve a workload without sidecar, we turn off injection in its namespace
+    // and also remove any override annotation for sidecar injection.
+    this.namespaceAutoInjectionEnabled = false;
+    this.workloadHasSidecar = false;
+    this.workloadHasAutoInjectionOverride = false;
+
+    // Make sure that injection in the namespace is turned off
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": null,
+                "istio.io/rev": null
+            }
+        }
+    });
+
+    // Make sure that the workload does not have override configuration
+    cy.request('PATCH', `/api/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}?type=Deployment`, {
+        spec: {
+            template: {
+                metadata: {
+                    labels: {
+                        "sidecar.istio.io/inject": null
+                    }
+                }
+            }
+        }
+    });
+
+    // Restart the workload to ensure changes are applied.
+    cy.exec('kubectl delete pod -n default -l app=sleep')
+});
+
+Given('a workload with a sidecar', function () {
```

Comment:
`function` is used here instead of arrow so that state can be shared with `this`?

Reply:
Yes :-)